eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -0,0 +1,471 @@
+. (Join-Path $PSScriptRoot SemVer.ps1)
+
+$VERSION_REGEX = "(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?((?<pre>[^0-9][^\s]+))?"
```

Comment:
Is this still needed when you pull in SemVer? If we are missing some versioning helpers maybe better to add them to SemVer.

Reply:
100%. (ha)

I'm now completely utilizing the SemVer.ps1 code. Notice that I've made a SLIGHT change to that to make the regex static so that people outside the class can leverage the resource.